pkg/component/component.go

```diff
@@ -402,6 +340,203 @@ func getS2IPaths(podEnvs []corev1.EnvVar) []string {
+		ApplicationName: appName,
+	}
+	createArgs.SourceType = cmpSrcType
+	createArgs.SourcePath = componentConfig.GetSourceLocation()
+
+	if len(cmpPorts) > 0 {
+		createArgs.Ports = cmpPorts
+	}
+
+	createArgs.Resources, err = occlient.GetResourceRequirementsFromCmpSettings(componentConfig)
+	if err != nil {
+		return errors.Wrap(err, "failed to create component")
+	}
+
+	switch cmpSrcType {
+	case config.GIT:
+		// Use Git
+		if cmpSrcRef != "" {
+			createArgs.SourceRef = cmpSrcRef
+		}
+		if err = CreateFromGit(
+			client,
+			createArgs,
+		); err != nil {
+			return errors.Wrapf(err, "failed to create component with args %+v", createArgs)
+		}
+		// Git is the only one using BuildConfig since we need to retrieve the git
+		if err = Build(client, createArgs.Name, createArgs.ApplicationName, true, stdout); err != nil {
+			return errors.Wrapf(err, "failed to build component with args %+v", componentConfig)
+		}
+	case config.LOCAL:
+		fileInfo, err := os.Stat(createArgs.SourcePath)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get info of path %+v of component %+v", createArgs.SourcePath, createArgs)
+		}
+		if !fileInfo.IsDir() {
+			return fmt.Errorf("component creation with args %+v as path needs to be a directory", createArgs)
+		}
+		// Create
+		if err = CreateFromPath(client, createArgs); err != nil {
+			return errors.Wrapf(err, "failed to create component with args %+v", createArgs)
+		}
+	case config.BINARY:
+		if err = CreateFromPath(client, createArgs); err != nil {
+			return errors.Wrapf(err, "failed to create component with args %+v", createArgs)
+		}
+	default:
+		// If the user does not provide anything (local, git or binary), use the current absolute path and deploy it
+		createArgs.SourceType = config.LOCAL
+		dir, err := os.Getwd()
+		if err != nil {
+			return errors.Wrapf(err, "cannot create component with current path as local source path since no component source details are passed")
+		}
+		createArgs.SourcePath = dir
+		if err = CreateFromPath(client, createArgs); err != nil {
+			return errors.Wrapf(err, "")
+		}
+	}
+	return
+}
+
+// ValidateComponentCreateRequest validates request for component creation and returns errors if any
+// Parameters:
+//	componentSettings: Component settings
+//	isCmpExistsCheck: boolean to indicate whether or not error out if component with same name already exists
+// Returns:
+//	errors if any
+func ValidateComponentCreateRequest(client *occlient.Client, componentSettings config.ComponentSettings, isCmpExistsCheck bool) (err error) {
+	// Check manddatory fields are set
+	if componentSettings.Name == nil ||
+		componentSettings.Application == nil ||
+		componentSettings.Project == nil ||
+		componentSettings.SourceType == nil ||
+		componentSettings.SourceLocation == nil ||
+		componentSettings.Type == nil {
+		return fmt.Errorf("component name, application, project, source type, source location and component type are mandatory parameters for component creation")
+	}
```

Comment:
I don't like what's happening above...

Why not change this into a separate function such as CheckMandatoryParameters that passes in `componentSettings` ?

And within that function, actually like what is required / needed. For example, if Type and SourceType is missing, it will error out saying "component type and source type is required... etc"

Reply:
we do this with `catalog.Exists` (separated into a another function for checks..)

Reply:
Ack.